manim/mobject/tabular.py

```diff
@@ -0,0 +1,1089 @@
+        element_to_mobject: Callable[
+            [Union[float, str, "VMobject"]], "VMobject"
+        ] = Paragraph,
+        element_to_mobject_config: Optional[dict] = {},
+        arrange_in_grid_config: Optional[dict] = {},
+        line_config: Optional[dict] = {},
+        **kwargs,
+    ):
+        """
+        Parameters
+        ----------
+        table
+            A 2d array or list of lists. Content of the table has to be a valid input
+            for the callable set in `element_to_mobject`.
+        row_labels
+            List of :class:`~.VMobject` representing labels of every row.
+        col_labels
+            List of :class:`~.VMobject` representing labels of every column.
+        top_left_entry
+            Top-left entry of the table, only possible if row and
+            column labels are given.
+        v_buff
+            Vertical buffer passed to :meth:`~.Mobject.arrange_in_grid`, by default 0.8.
+        h_buff
+            Horizontal buffer passed to :meth:`~.Mobject.arrange_in_grid`, by default 1.3.
+        include_outer_lines
+            `True` if should include outer lines, by default False.
+        add_background_rectangles_to_entries
+            `True` if should add backgraound rectangles to entries, by default False.
+        include_background_rectangle
+            `True` if should include background rectangle, by default False.
+        element_to_mobject
+            Element to mobject, by default :class:`~.Paragraph`. For common choices, see :mod:`~.text_mobject`
+            and :mod:`~.tex_mobject`.
+        element_to_mobject_config
+            Element to mobject config, by default {}.
+        arrange_in_grid_config
+            Dict passed to :meth:`~.Mobject.arrange_in_grid`, customizes the arrangement of the table.
+        line_config
+            Dict passed to :class:`~.Line`, customizes the lines of the table.
+        kwargs : Any
+            Additional arguments to be passed to :class:`~.VGroup`.
+        """
+
+        self.row_labels = row_labels
+        self.col_labels = col_labels
+        self.top_left_entry = top_left_entry
+        self.row_dim = len(table)
+        self.col_dim = len(table[0])
+        self.v_buff = v_buff
+        self.h_buff = h_buff
+        self.include_outer_lines = include_outer_lines
+        self.add_background_rectangles_to_entries = add_background_rectangles_to_entries
+        self.include_background_rectangle = include_background_rectangle
+        self.element_to_mobject = element_to_mobject
+        self.element_to_mobject_config = element_to_mobject_config
+        self.arrange_in_grid_config = arrange_in_grid_config
+        self.line_config = line_config
+        for row in table:
+            if len(row) == len(table[0]):
+                pass
+            else:
+                raise ValueError("Not all rows in table have the same length.")
+        VGroup.__init__(self, **kwargs)
+        mob_table = self.table_to_mob_table(table)
```

Comment:
It'd be helpful if there were some comments/organization to make it easier to parse what's happening here.

Reply:
You mean in total or a specific line you have in mind?

Reply:
I think now that the private methods are documented this isn't necessary 👍